src/src/beneficiaries/beneficiaries.service.ts

```diff
@@ -297,10 +305,276 @@ if(!result){
   remove(id: number) {
     // return this.hasuraService.delete(this.table, { id: +id });
   }
+  
  public async statusUpdate(req:any){
-    console.log("req",req)
   return await this.hasuraService.update(req.id, 'beneficiaries',req,this.returnFields,[...this.returnFields,"id"]);
 
   }
-}
 
+  public async registerBeneficiary(body, request) {
+    const axios = require('axios');
+    const user=await this.userService.ipUserInfo(request);
+    const password = body.mobile;
+    let username = body.first_name;
+    username += `_${body.mobile}`;
+    const data_to_create_user = {
+      enabled: 'true',
+      firstName: body.first_name,
+      username: username.toLowerCase(),
+      credentials: [
+        {
+          type: 'password',
+          value: password,
+          temporary: false,
+        },
+      ],
+      groups: ['beneficiaries'],
+    };
+    const adminResult = await this.helper.getAdminKeycloakToken();
+    
+    if (adminResult?.data?.access_token) {
```

Comment:
this keycloak service is already written in services/keycloak.service.ts Please import this service from there.